ianshulx/React-projects-for-beginners
# FunFact Generator and HEX/RGB CONVERTER apps added

**Title:** Add Fun Fact Generator Component with Real-Time API Integration

**Description:** This PR adds a Fun Fact Generator component built with Vite and React. The component fetches random fun facts from an external API and displays them in a user-friendly interface, making it a lighthearted and informative feature.

**Key Features:**
Real-Time Fun Fact Fetching: Retrieves a new fun fact from the [uselessfacts.jsph.pl](**https://uselessfacts.jsph.pl/**) API each time the user clicks the button.
Loading Indicator: Displays a loading message while the fact is being fetched to improve user experience.
Styled UI: Uses Tailwind CSS to create a clean, responsive, and accessible design.

**How to Test:**
Clone the repository and navigate to the project directory.
Run **npm install** to install dependencies.
Start the app using **npm run dev**.
Click "Get Fun Fact" to fetch and display random fun facts.


**Title:** Add HEX/RGB Color Converter Component for Easy Color Conversion

**Description:** This PR introduces a HEX/RGB Color Converter component built with Vite and React. The component allows users to convert colors between HEX and RGB formats, making it a convenient tool for designers and developers.

**Key Features:**
Two-Way Conversion: Supports conversions from HEX to RGB and RGB to HEX.
User-Friendly Input Validation: Provides feedback for invalid color formats, ensuring smooth and error-free usage.
Styled Interface: Uses Tailwind CSS for a responsive and accessible design, displaying the converted color with a live preview.

**How to Test:**
Clone the repository and navigate to the project directory.
Run **npm install** to install dependencies.
Start the app using **npm run dev**.
Enter a HEX color (e.g., #FF5733) or RGB color (e.g., rgb(255, 87, 51)) and click "Convert" to see the conversion.
